llvm/lib/Target/PowerPC/PPCAsmPrinter.cpp

```diff
@@ -846,6 +870,15 @@ void PPCAsmPrinter::emitInstruction(const MachineInstr *MI) {
       return MCSymbolRefExpr::VariantKind::VK_PPC_AIX_TLSGDM;
     if (MO.getTargetFlags() & PPCII::MO_TLSGD_FLAG)
       return MCSymbolRefExpr::VariantKind::VK_PPC_AIX_TLSGD;
+    if (MO.getTargetFlags() & PPCII::MO_TLSLD_FLAG) {
+      if (isSpecialAIXSymbolTLSML(MO, IsAIX))
```

Comment:
Why not just compare name at the beginning rather than checking TargetFlag?

Reply:
The symbol is specific for TLSLD, so I prefer to encapsulate the logic inside TLSLD relevant predator.

Reply:
```suggestion
      if (isAIX && isTLSMLSymbolOperand(MO))
```
`bool` in function's argument list is not preferred.

Reply:
Oh, I saw in https://llvm.org/docs/ProgrammersManual.html there is function which take bool as argument. I won't change unless somebody change that document.